src/Lykke.Job.CandlesHistoryWriter.Services/HistoryMigration/MidPriceFixManager.cs

```
@@ -0,0 +1,255 @@
+            });
+
+            return CandlesFiltrationManager.FiltrationLaunchResult.Started;
+        }
+
+        private async Task FixMidPricesAsync(string assetPairId, int assetPairAccuracy, bool analyzeOnly)
+        {
+            try
+            {
+                _log.Info(nameof(FixMidPricesAsync), $"Starting mid candle prices fix for {assetPairId}...");
+
+                Health.Message = "Getting candles to fix...";
+                var candles = await TryGetCorruptedCandlesAsync(assetPairId, assetPairAccuracy);
+
+                if (!candles.Any())
+                {
+                    _log.Info(nameof(FixMidPricesAsync),
+                        $"There are no candles to fix for {assetPairId}. Skipping.");
+                    return;
+                }
+
+                if (analyzeOnly)
+                {
+                    _log.Info(nameof(FixMidPricesAsync),
+                        $"Mid candle prices fix for {assetPairId} finished: analyze only. Candles to fix: {Health.CorruptedCandlesCount}");
+                    return;
+                }
+
+                //sort by interval
+                var candlesByInterval = candles
+                    .GroupBy(c => (int)c.TimeInterval)
+                    .ToSortedDictionary(g => g.Key);
+
+                Health.Message = "Fixing candles from Sec to bigger intervals...";
+
+                //replace candles from Sec to bigger intervals
+                foreach (var candleBatch in candlesByInterval)
+                {
+                    var candlesToReplace = candleBatch.Value.ToList();
+                    await _candlesHistoryRepository.ReplaceCandlesAsync(candlesToReplace);
+                    Health.FixedCandlesCount += candlesToReplace.Count;
+                }
+
+                _log.Info(nameof(FixMidPricesAsync),
+                    $"Mid candle prices fix for {assetPairId} finished.");
+            }
+            catch (Exception ex)
```

Why do we swallow exception here?